airflow/models.py

```diff
@@ -797,8 +797,23 @@ class TaskInstance(Base, LoggingMixin):
     def __init__(self, task, execution_date, state=None):
         self.dag_id = task.dag_id
         self.task_id = task.task_id
-        self.execution_date = execution_date
         self.task = task
+        self._log = logging.getLogger("airflow.task")
+
+        # make sure we have a localized execution_date stored in UTC
+        if execution_date and not timezone.is_localized(execution_date):
```

Comment:
An execution date should be not-null, right?

Reply:
Some of the tests assume it can be, unfortunately (All the task dependency tests)

Reply:
Ok, maybe something to look into, but that's a different issue then.